### PR TITLE
BFD-2869: Removal of ACLs on S3 terraservices

### PR DESCRIPTION
**Pull request by brandoncruz3**
**JIRA Ticket:**
[BFD-2869](https://jira.cms.gov/browse/BFD-2869)

**User Story or Bug Summary:**

As a BFD infrastructure engineer, I want the solutions for AWS S3 bucket management encoded in terraform to have fewer exceptions so that I can better maintain the infrastructure-as-code solution.

AC:
* ACL and logging configuration for pipeline terraservice S3 bucket (ETL bucket) definitions are reviewed for accuracy
* Non-default ACLs, and logging policies are updated and/or removed as appropriate
* pipeline terraservice's s3.tf FIXME comments are eliminated

---

### What Does This PR Do?

Drops ACLs from S3 terraservice buckets and removal/updates FIXMEs

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/terraform/services/common/s3.tf [hunk 1]
```diff
@@ -8,16 +8,6 @@ resource "aws_s3_bucket" "this" {
...
-# block public access to the bucket
-resource "aws_s3_bucket_public_access_block" "this" {
-  bucket = aws_s3_bucket.this.id
-
-  block_public_acls       = true
-  block_public_policy     = true
-  ignore_public_acls      = true
-  restrict_public_buckets = true
-}
-
```
Comment (malessi) [inline ops/terraform/services/common/s3.tf:20]: I'm not sure if we want to remove these public access blocks? While the `aws_s3_bucket_acl` resources are basically deprecated with the recent default to disabling ACLs altogether, I don't believe this resource is deprecated in the same way. At worst, there's a bit of _explicit_ redundancy with this `aws_s3_bucket_public_access_block`, and I'm not sure what the behavior is if we _remove_ this resource for Buckets that have existed prior to the April change.

Comment (brandoncruz3) [reply ops/terraform/services/common/s3.tf]: Thanks! Added back in.

File: ops/terraform/services/common/s3.tf [hunk 2]
```diff
@@ -29,27 +28,16 @@ resource "aws_s3_bucket_server_side_encryption_configuration" "this" {
...
 
 resource "aws_s3_bucket_logging" "this" {
-  # TODO: Make this work better for ephemeral environments, etc
   count = local.is_ephemeral_env ? 0 : 1
 
   bucket = aws_s3_bucket.this.id
 
-  # TODO: consider adding this...
-  # expected_bucket_owner = local.account_id
+  expected_bucket_owner = local.account_id
```
Comment (keithdadkins) [inline ops/terraform/services/common/s3.tf:35]: This is a good change; but, while I do not expect we would be shipping logs across accounts, I'm not 100% certain we won't be. (ie. multiple aws accounts per environment with our upcoming v4/cloudtamer migration).

Comment (keithdadkins) [reply ops/terraform/services/common/s3.tf:35]: Not requesting any changes here, just something to note.

**Review by keithdadkins — APPROVED** [review]

File: ops/terraform/services/common/s3.tf (hunk 2 above)
Comment (brandoncruz3) [reply ops/terraform/services/common/s3.tf:35]: If you want, I can switch this back and we can circle back when we get to working on v4 migration.

Comment (keithdadkins) [reply ops/terraform/services/common/s3.tf:35]: No it's ok. Just something to note during our migration. Doubt it will come up.

**Review by brianburton — APPROVED** [review]

**Pull request merged**